prometheus-surfboard-exporter: init at 2.0.0

disassembler opened the pull request:
###### Motivation for this change
Haven't tested the nixos module yet (will try to tomorrow), but I tested running the exporter standalone and a curl of the metric page got me all my surfboard cable modem stats.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
prometheus-surfboard-exporter: init at 2.0.0

Commit:
nixos/prometheus-surfboard-exporter: add new module

fpletz reviewed (approved):
Code looks fine to me. Thanks!

Mic92 commented:
nitpick for commit message:

```
nixos/prometheus-surfboard-exporter: add service
```

Mic92 commented on nixos/modules/services/monitoring/prometheus/surfboard-exporter.nix:
```diff
@@ -0,0 +1,67 @@
+        User = "nobody";
+        Restart = "always";
+        PrivateTmp = true;
+        WorkingDirectory = /tmp;
```
Does this application creates temporary files securely? There might be other applications racing for `/tmp`. Also a dedicated user different from `nobody` would be helpful. Unfortunately we do not have the latest systemd release yet.
Otherwise runtime directories could be create automatically for `DynamicUser`.

disassembler commented:
Looking at the code, doesn't look like it writes any files. Everything is stdout or the http socket opened. I've removed the work dir and private tmp attributes. I also fixed the commit message on service commit to prefix with nixos/

Mic92 commented on nixos/modules/services/monitoring/prometheus/surfboard-exporter.nix:
```diff
@@ -0,0 +1,65 @@
+      unitConfig.Documentation = "https://github.com/ipstatic/surfboard_exporter";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        User = "nobody";
```
The following should also work here then:

```
DynamicUser = "yes";
```

orivej commented on pkgs/servers/monitoring/prometheus/surfboard-exporter.nix:
```diff
@@ -0,0 +1,24 @@
+buildGoPackage rec {
+  name = "surfboard_exporter-${version}";
+  version = "2.0.0";
+  rev = version;
```
This line is redundant.

WilliButz commented:
@disassembler I adjusted [the commit that adds the module](https://github.com/WilliButz/nixpkgs/commits/surfboard-pr-disasm), to work with the new prometheus exporters module (and also rebased this on the current master).

Mic92 commented:
@WilliButz can you open a pull request?

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.31645&attempt_id=17b5c7dd-fd11-4b2c-b460-c204958080ca)

Attempted: prometheus-surfboard-exporter

<details><summary>Partial log (click to expand)</summary><p>

```
strip is /nix/store/j75dgadrff2d1fyc4fczmcgqkid2imdx-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/x9d90m3pyqsdq73dkrx3iq8yq7p1isqb-surfboard_exporter-2.0.0-bin/bin
patching script interpreter paths in /nix/store/x9d90m3pyqsdq73dkrx3iq8yq7p1isqb-surfboard_exporter-2.0.0-bin
checking for references to /build in /nix/store/x9d90m3pyqsdq73dkrx3iq8yq7p1isqb-surfboard_exporter-2.0.0-bin...
shrinking RPATHs of ELF executables and libraries in /nix/store/xjnikwhsq09fk262pcjfay797wxv67mr-surfboard_exporter-2.0.0
strip is /nix/store/j75dgadrff2d1fyc4fczmcgqkid2imdx-binutils-2.30/bin/strip
patching script interpreter paths in /nix/store/xjnikwhsq09fk262pcjfay797wxv67mr-surfboard_exporter-2.0.0
/nix/store/xjnikwhsq09fk262pcjfay797wxv67mr-surfboard_exporter-2.0.0/share/go/src/github.com/ipstatic/surfboard_exporter/after_success.sh: interpreter directive changed from "/bin/bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
checking for references to /build in /nix/store/xjnikwhsq09fk262pcjfay797wxv67mr-surfboard_exporter-2.0.0...
/nix/store/x9d90m3pyqsdq73dkrx3iq8yq7p1isqb-surfboard_exporter-2.0.0-bin
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.31645&attempt_id=fc38a883-6f8a-4e9e-9b8f-b43eb4944267)

Attempted: prometheus-surfboard-exporter

<details><summary>Partial log (click to expand)</summary><p>

```
/private/tmp/nix-build-surfboard_exporter-2.0.0.drv-0/go /private/tmp/nix-build-surfboard_exporter-2.0.0.drv-0
/private/tmp/nix-build-surfboard_exporter-2.0.0.drv-0
post-installation fixup
strip is /nix/store/kdff2gim6417493yha769kh00n63lnrw-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/zbix4n1ipswnya1l7shp67r0d9ipwq53-surfboard_exporter-2.0.0-bin/bin
patching script interpreter paths in /nix/store/zbix4n1ipswnya1l7shp67r0d9ipwq53-surfboard_exporter-2.0.0-bin
strip is /nix/store/kdff2gim6417493yha769kh00n63lnrw-cctools-binutils-darwin/bin/strip
patching script interpreter paths in /nix/store/qy8ck7rf6prk1zc98p04c1naill576xc-surfboard_exporter-2.0.0
/nix/store/qy8ck7rf6prk1zc98p04c1naill576xc-surfboard_exporter-2.0.0/share/go/src/github.com/ipstatic/surfboard_exporter/after_success.sh: interpreter directive changed from "/bin/bash" to "/nix/store/r8bx3qf1bpncb14i9gzma4vr089pc3pv-bash-4.4-p19/bin/bash"
/nix/store/zbix4n1ipswnya1l7shp67r0d9ipwq53-surfboard_exporter-2.0.0-bin
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.31645&attempt_id=92d790d3-b116-4582-9b40-a0134ffe9a64)

Attempted: prometheus-surfboard-exporter

<details><summary>Partial log (click to expand)</summary><p>

```
strip is /nix/store/gp7fylxwn18b7pl2c18ks89hsiaxyfvf-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/n34lzfiky5aviycxfy5g27azgnrj5qpr-surfboard_exporter-2.0.0-bin/bin
patching script interpreter paths in /nix/store/n34lzfiky5aviycxfy5g27azgnrj5qpr-surfboard_exporter-2.0.0-bin
checking for references to /build in /nix/store/n34lzfiky5aviycxfy5g27azgnrj5qpr-surfboard_exporter-2.0.0-bin...
shrinking RPATHs of ELF executables and libraries in /nix/store/5anw43shi54yjz9x2h8c6xgmfa7xgvna-surfboard_exporter-2.0.0
strip is /nix/store/gp7fylxwn18b7pl2c18ks89hsiaxyfvf-binutils-2.30/bin/strip
patching script interpreter paths in /nix/store/5anw43shi54yjz9x2h8c6xgmfa7xgvna-surfboard_exporter-2.0.0
/nix/store/5anw43shi54yjz9x2h8c6xgmfa7xgvna-surfboard_exporter-2.0.0/share/go/src/github.com/ipstatic/surfboard_exporter/after_success.sh: interpreter directive changed from "/bin/bash" to "/nix/store/a4qslf0yfs44mhnbagarn232sjnckhy0-bash-4.4-p19/bin/bash"
checking for references to /build in /nix/store/5anw43shi54yjz9x2h8c6xgmfa7xgvna-surfboard_exporter-2.0.0...
/nix/store/n34lzfiky5aviycxfy5g27azgnrj5qpr-surfboard_exporter-2.0.0-bin
```
</p></details>
